File: go/226_Invert_Binary_Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
    if (root == nil) {
        return root
    }
    var left = invertTree(root.Left)
    var right = invertTree(root.Right)
    root.Right = left
    root.Left = right
    return root
}

File: kotlin/226_Invert_Binary_Tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null) {
            return root;
        }
        var left = invertTree(root.left);
        var right = invertTree(root.right);
        root.right = left;
        root.left = right;
        return root;
    }
}